pandas/core/sorting.py

```diff
@@ -320,7 +321,12 @@ def lexsort_indexer(
     keys = [ensure_key_mapped(k, key) for k in keys]
 
     for k, order in zip(keys, orders):
-        cat = Categorical(k, ordered=True)
+        with warnings.catch_warnings():
```

Comment:
hmm does this have any perf impact?

Reply:
```
df = pd.DataFrame({"A": range(5), "B": range(5), "C": range(5)})

%timeit df.sort_values(["A", "B"])
486 µs ± 10.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)  # <- main
527 µs ± 28 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)  # <- PR
```

If we bump the length up to 10**5 its 21.7 ms vs 21.9 ms

Reply:
ok so nothing